weldx/transformations/local_cs.py

```diff
@@ -690,6 +734,38 @@ def as_rotation(self) -> Rot:  # pragma: no cover
         """
         return Rot.from_matrix(self.orientation.values)
 
+    def as_homogeneous_matrix(self, translation_unit: UnitLike) -> np.ndarray:
+        """Get a homogeneous transformation matrix from the coordinate system
+        orientation.
+
+        Parameters
+        ----------
+        translation_unit : UnitLike
+            Unit the translation part of the homogeneous transformation matrix
+            should represent.
+
+        Returns
+        -------
+        numpy.ndarray
+            Numpy array representing the homogeneous transformation matrix.
+
+        """
+
+        if self.is_time_dependent:
+            time_dim = self.time.shape[0]
+        else:
+            time_dim = 1
+
+        rotation = np.resize(self.orientation.data, (time_dim, 3, 3))
+        translation = np.resize(
+            self.coordinates.data.to(translation_unit).m, (time_dim, 3)
+        )
```

Comment:
Apparently this expression raises a mypy error, because using `self.coordinates` could return a `TimeSeries`, which in turn would return `pint.Quantity | MathematicalExpression` when `.data` is accessed and the latter one does not support unit transformations. Not sure how to fix this.

Reply:
I think we can leave out supporting `TimeSeries` for now (throwing an error).

We could add
```py
if isinstance(self.coordinates, TimeSeries):
    raise NotImplementedError("Cannot convert LCS with `TimeSeries` coordinates to homogeneous matrix")
```
at the beginning